packages/accounts/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+//! manages the composition of authorization intents from various sources. It
+//! separates the "who" (signers), "what" (rules for specific actions), and
+//! "how" (pluggable policies) of authorization, enabling flexible combinations
+//! of authentication methods.
+//!
+//! Smart accounts can combine multiple authorization mechanisms seamlessly. For
+//! example, a wallet might require both a session policy and a passkey with a
+//! 24-hour lifespan, treating them as a single composite "key" for signing
+//! actions.
+//!
+//! ## Core Components
+//!
+//! ### Context Rules
+//!
+//! Context Rules define the scope of authorization and compose signers with
+//! policies:
+//!
+//! - Set authorization requirements for specific context types (contract calls,
+//!   deployments)
+//! - Must contain at least one signer or one policy
+//! - Support optional expiration via ledger sequence (`valid_until`)
+//! - Allow multiple rules for the same context with different configurations
+//! - Evaluated in reverse chronological order (most recent first)
+//!
+//! Supported context types:
+//! - `Default`: Applies to any context
+//! - `CallContract(Address)`: Specific contract calls
+//! - `CreateContract(BytesN<32>)`: Contract deployments
+//!
+//! ### Signers
+//!
+//! Two types of signers provide authentication:
+//!
+//! #### Native Signers
+//! - Any Soroban `Address` that can be authorized
+//! - Verified within the smart account using `require_auth_for_args()`
+//! - Requires manual construction of authorization entries during simulation
+//!
+//! #### Delegated Signers
+//! - Tuple of `(Address, Bytes)`: verifier contract + public key
+//! - Offloads signature verification to trusted, immutable verifier contracts
+//! - Supports multiple cryptographic schemes without modifying the account
+//! - Enables future authentication methods (zk-based, email signing)
+//!
+//! ### Policies
+//!
+//! Policies customize and enforce signer behavior within context rules:
+//!
+//! - External contracts with standardized interfaces
+//! - Logic owned solely by the smart account
+//! - Support both per-account and shared configurations
+//! - All policies in a rule must be enforceable for authorization
+//!
+//! Policy interface:
+//! - `can_enforce()`: Pre-check without state changes
+//! - `enforce()`: Execute with potential state changes (account-only
+//!   invocation)
+//! - `install()`: Store custom configuration for account/context
+//! - `uninstall()`: Clean up stored configuration
+//!
+//! ## Authorization Flow
+//!
+//! The matching algorithm for authorization:
+//!
+//! 1. Retrieve all non-expired rules for the context type, plus defaults
+//! 2. For each rule (newest first):
+//!    - Filter authenticated signers from rule's signer set
+//!    - If policies exist: verify all are enforceable
+//!    - If no policies: require all signers authenticated
+//!    - First matching rule succeeds
+//! 3. If no rule matches, authorization fails
```

Comment:
so if only 1 rule matches, authorization succeeds? I don't think so?

Maybe something like: "In order for authorization to pass, all rules must pass"

Reply:
no, on the contrary: on the 1st rule that's satisfied, we consider the context validated